### PR TITLE
Update zfs destroy script

### PR DESCRIPTION
**Pull request by heary-cao**
issues:
update zfs destroy contents script

Solution:
update **zfs_destroy_008_pos.ksh zfs_destroy_009_pos.ksh zfs_destroy_010_pos.ksh zfs_destroy_011_pos.ksh 
zfs_destroy_012_pos.ksh zfs_destroy_013_neg.ksh** 

Signed-off-by: cao.xuewen <cao.xuewen@zte.com.cn>

### CODE REVIEW

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_common.kshlib [hunk 1]
```diff
@@ -97,7 +97,7 @@ function setup_testenv #[dtst]
 #
 function cleanup_testenv
 {
-	if is_global_zone && ismounted "$TESTDIR1" "ufs" ; then
+	if is_global_zone && ! ismounted "$TESTDIR1" "zfs" ; then
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_common.kshlib]: For portability `NEWFS_DEFAULT_FS` in place of "ufs".  It's defined to be "ext2" for Linux and is commonly used when a non-zfs filesystem needs to be created on a volume.

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_008_pos.ksh [hunk 2]
```diff
@@ -38,14 +38,27 @@
...
 for snap in $FSSNAP $VOLSNAP; do
-    log_must $ZFS destroy -d $snap
-    log_mustnot datasetexists $snap
+    if [[ $snap == $VOLSNAP ]]; then
+		if is_global_zone; then
+			test_s_run $snap
+		fi
+	else
+		test_s_run $snap
+	fi
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_008_pos.ksh]: Can you explain what exactly the test failure was and why this is needed?  At the moment under Linux `is_global_zone` should always be true.  The concept of illumos zones hasn't been mapped to Linux style containers for the test suite.  Given that I don't see any functional change here.

However, if one were to run the test suite in a container I could see how that would be a problem.

**Comment by heary-cao** [comment]
@behlendorf   At the moment under Linux is_global_zone should always be true. I agree it. but from the purpose of zfs_destroy_common.kshlib designers, is_global_zone may be false,  that is need to differentiate between true or false, so in order to code logic is normal, to distinguish in the footsteps of independent.

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/cleanup.ksh [hunk 3]
```diff
@@ -27,6 +27,6 @@
 
 . $STF_SUITE/tests/functional/cli_root/zfs_destroy/zfs_destroy_common.kshlib
 
-cleanup_testenv
+#cleanup_testenv
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/cleanup.ksh]: Why was it necessary to comment out this cleanup?

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_common.kshlib [hunk 4]
```diff
@@ -97,7 +97,7 @@ function setup_testenv #[dtst]
 #
 function cleanup_testenv
 {
-	if is_global_zone && ismounted "$TESTDIR1" "ufs" ; then
+	if is_global_zone && ! ismounted "$TESTDIR1" "ufs" ; then
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_common.kshlib]: You should use NEWFS_DEFAULT_FS here to cleanup the local ext2 filesystem created in setup_testenv() with NEWFS.

```
-	if is_global_zone && ismounted "$TESTDIR1" "ufs" ; then
+-	if is_global_zone && ismounted "$TESTDIR1" "$NEWFS_DEFAULT_FS" ; then
```

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_008_pos.ksh [hunk 5]
```diff
@@ -38,14 +38,27 @@
 #
 ################################################################################
 
+function test_s_run
+{
+	typeset snap=$1
+
+    log_must $ZFS destroy -d $snap
+    log_mustnot datasetexists $snap
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_008_pos.ksh]: Minor cstyle issue.  The originally test indented by 4 spaces but should have used tabs like the other test cases.  The same issue exists in several other of these tests.

File: tests/runfiles/linux.run [hunk 6]
```diff
@@ -97,15 +97,11 @@ tests = ['zfs_create_001_pos', 'zfs_create_002_pos', 'zfs_create_003_pos',
...
-# zfs_destroy_010_pos - busy mountpoint behavior
-# zfs_destroy_011_pos - busy mountpoint behavior
-# zfs_destroy_012_pos - busy mountpoint behavior
-# zfs_destroy_013_neg - busy mountpoint behavior
 [tests/functional/cli_root/zfs_destroy]
 tests = ['zfs_destroy_002_pos', 'zfs_destroy_003_pos', 'zfs_destroy_004_pos',
-    'zfs_destroy_006_neg', 'zfs_destroy_007_neg', 'zfs_destroy_014_pos',
+    'zfs_destroy_006_neg', 'zfs_destroy_007_neg', 'zfs_destroy_008_pos',
+	'zfs_destroy_009_pos', 'zfs_destroy_010_pos', 'zfs_destroy_011_pos',
+	'zfs_destroy_012_pos', 'zfs_destroy_013_neg', 'zfs_destroy_014_pos',
```
Comment (behlendorf) [inline tests/runfiles/linux.run]: Minor style issue, fix indent.  Continued lines have a 4 space indent.

**Comment by behlendorf** [comment]
> that is need to differentiate between true or false, so in order to code logic is normal.

OK, I just wanted to make sure that's why you needed to make this change.  Please clearly explain that in the commit comment, I'm sure other developers will have the same question.  I've provided a few additional review comments.

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/cleanup.ksh (hunk 3 above)
Comment (heary-cao) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/cleanup.ksh]: @behlendorf 
zfs_destroy test suite, Each individual destroy test case, All have a cleanup_testenv when the test case is log_onexit ,  For example log_onexit cleanup_testenv,  Therefore, zfs_destroy cleanup.ksh  again execute cleanup_testenv is unnecessary.

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_008_pos.ksh (hunk 2 above)
Comment (heary-cao) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_008_pos.ksh]: At the moment under Linux is_global_zone should always be true. I agree it. but from the purpose of zfs_destroy_common.kshlib designers, is_global_zone may be false, that is need to differentiate between true or false, so in order to code logic is normal, to distinguish in the footsteps of independent.
Therefore, When we zfstestsuite test VOL or VOL snapshot or VOL clone, must consider is_global_zone value,

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_common.kshlib [hunk 7]
```diff
@@ -97,7 +97,7 @@ function setup_testenv #[dtst]
 #
 function cleanup_testenv
 {
-	if is_global_zone && ismounted "$TESTDIR1" "ufs" ; then
+	if is_global_zone && ! ismounted "$TESTDIR1" "$NEWFS_DEFAULT_FS" ; then
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_common.kshlib]: I believe you have a stray `!` here.  We want to cleanup the test ext2 filesystems on cleanup if they're mounted.

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/cleanup.ksh (hunk 3 above)
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/cleanup.ksh]: Then let's simply remove it rather than comment it out.

Comment (heary-cao) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/cleanup.ksh]: @behlendorf ok！

**Pull request merged**